### DISCUSSION TITLE
Left navigation: Categories - unfold only unread

### DISCUSSION THREAD

**Discussion by shalak** [FreshRSS/FreshRSS#5452] [Ideas]
It would be great, to have the Categories pane on left navigation fold/unfold in three modes:
- completely folded, i.e. show none feeds in this category
- completely unfolded, i.e. show all  feeds in this category
- new mode: show only those feeds that contains unread articles

See [my comment](https://github.com/FreshRSS/FreshRSS/discussions/5452#discussioncomment-6137498) for the proof of concept.

**Comment by Frenzie** [comment]
I'm not entirely sure I understand. Isn't that how it already works? (With "show unread" enabled and "show read" disabled.)

**Reply by shalak** [reply]
Where are those options you mention? I only see this:
<img width="902" alt="image" src="https://github.com/FreshRSS/FreshRSS/assets/1536554/32ecbed7-566a-447a-b526-7bdc55dd71fe">

**Reply by shalak** [reply]
I can only toggle between to modes, all visible:

<img width="278" alt="image" src="https://github.com/FreshRSS/FreshRSS/assets/1536554/1071a710-0d46-46b5-8f6b-2366741b6abe">

Or none visible:

<img width="272" alt="image" src="https://github.com/FreshRSS/FreshRSS/assets/1536554/a75dd4c4-db62-4159-b359-4ab9308d244b">


I'd like have a mode of unfolding in which I'd only see the Wikipedia feed and the FreshRSS feed is hidden

**Reply by Frenzie** [reply]
I'm referring to the top bar.
![image](https://github.com/FreshRSS/FreshRSS/assets/202757/6c8b428a-c7df-4a5e-9df8-90c0be7b1e52)

**Reply by shalak** [reply]
Those refer to the articles on given feed/category. I mean the view on the left hand side, in the navigation page. See the pictures in my previous post to see what I mean.

**Reply by shalak** [reply]
I added a comment to main thread with the PoC of the solution

**Reply by Frenzie** [reply]
> Those refer to the articles on given feed/category. I mean the view on the left hand side, in the navigation page.

As do I.

**Reply by shalak** [reply]
Selecting those icons on the top bar have no effect on the left navigation pane, at least in my installation.

**Reply by Frenzie** [reply]
That would be related to this setting in Configuration → Reading.
![image](https://github.com/FreshRSS/FreshRSS/assets/202757/8af51967-b3c7-4595-a12c-7438d7ac1b7b)

**Reply by shalak** [reply]
Configuration → Reading doesn't allow to change a way the category is unfolded. Only to choose which categories to completely unfold. [Here](https://github.com/FreshRSS/FreshRSS/discussions/5452#discussioncomment-6137498) is what I mean.

**Comment by shalak** [comment]
Here's the proof of concept, a rather dirty hack that introduces the logic I'm talking about (I'm not a frontend engineer, don't shoot 🫣):

```diff
diff --git a/p/scripts/main.js b/p/scripts/main.js
index a75a731c..9ba75120 100644
--- a/p/scripts/main.js
+++ b/p/scripts/main.js
@@ -818,14 +818,36 @@ function init_column_categories() {
         if (a) {
             const icon = a.querySelector('.icon');
             const category_id = a.closest('.category').id;
+            const ul = a.closest('li').querySelector('.tree-folder-items');
+
             if (icon.alt === '🔽' || icon.innerHTML === '🔽') {
                 if (icon.src) {
-                    icon.src = icon.src.replace('/icons/down.', '/icons/up.');
+                    icon.src = icon.src.replace('/icons/down.', '/icons/right.');
+                    icon.alt = '▶️';
+                } else {
+                    icon.innerHTML = '▶️';
+                }
+                Array.from(ul.children).forEach(li => {
+                    if (parseInt(li.getAttribute('data-unread')) > 0) {
+                        li.style.display = '';
+                    } else {
+                        li.style.display = 'none';
+                    }
+                });
+                ul.classList.add('active');
+                rememberOpenCategory(category_id, true);
+            } else if (icon.alt === '▶️' || icon.innerHTML === '▶️') {
+                if (icon.src) {
+                    icon.src = icon.src.replace('/icons/right.', '/icons/up.');
                     icon.alt = '🔼';
                 } else {
                     icon.innerHTML = '🔼';
                 }
+                ul.classList.remove('active');
                 rememberOpenCategory(category_id, true);
+                Array.from(ul.children).forEach(li => {
+                        li.style.display = 'block';
+                });
             } else {
                 if (icon.src) {
                     icon.src = icon.src.replace('/icons/up.', '/icons/down.');
@@ -833,17 +855,20 @@ function init_column_categories() {
                 } else {
                     icon.innerHTML = '🔽';
                 }
+                ul.classList.add('active');
                 rememberOpenCategory(category_id, false);
+                Array.from(ul.children).forEach(li => {
+                        li.style.display = 'block';
+                });
+
             }
 
-            const ul = a.closest('li').querySelector('.tree-folder-items');
             let nbVisibleItems = 0;
             for (let i = ul.children.length - 1; i >= 0; i--) {
                 if (ul.children[i].offsetHeight) {
                     nbVisibleItems++;
                 }
             }
-            ul.classList.toggle('active');
             // CSS transition does not work on max-height:auto
             ul.style.maxHeight = ul.classList.contains('active') ? (nbVisibleItems * 4) + 'em' : 0;
             return false;
```

Corresponding `right.svg` file:

```xml
<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16">
  <path fill="#666" d="M215.03 759v-1.031a1.039 1.039 0 0 0-.313-.688L209 750.984l-5.72 6.297a1.014 1.014 0 0 0-.28.72v1h1c.265-.001.53-.094.718-.282l4.282-4.83 4.28 4.83c.195.21.47.303.75.281z" color="#bebebe" style="block-progression:tb;text-indent:0;text-align:start;text-transform:none" transform="translate(-201 -747) rotate(90 207.5 753.5)"/>
</svg>
```


Here's how it looks:

https://github.com/FreshRSS/FreshRSS/assets/1536554/fbb44885-f434-46ce-bbaf-f129139c7355
